modyn/evaluator/internal/metric_factory.py

```diff
@@ -1,13 +1,38 @@
-from modyn.config.schema.pipeline import validate_metric_config_json
-from modyn.evaluator.internal.metrics import AbstractEvaluationMetric, AbstractHolisticMetric, Accuracy, F1Score, RocAuc
+from typing import overload
+
+from modyn.config.schema.pipeline.evaluation.metrics import (
+    MetricConfig,
+    validate_metric_config_json,
+)
+from modyn.evaluator.internal.metrics import (
+    AbstractEvaluationMetric,
+    AbstractHolisticMetric,
+    Accuracy,
+    F1Score,
+    RocAuc,
+)
 
 all_metrics: set[type[AbstractEvaluationMetric]] = {Accuracy, F1Score, RocAuc}
 
 
 class MetricFactory:
+    @overload
```

Comment:
Also support calling with pydantic models and not just their json-string equivalents.